### PR TITLE
feat: Move client.InjectHTTPClient to the client.Config struct

### PR DESCRIPTION
**Pull request by Mehrbod2002**
Fixes : #398 
Move method to client.Config

### CODE REVIEW

File: core/endpoint_test.go [hunk 1]
```diff
@@ -17,7 +17,8 @@ import (
 )
 
 func TestEndpoint(t *testing.T) {
-	defer client.InjectHTTPClient(nil)
+	config := client.GetDefaultConfig()
```
Comment (TwiN) [inline core/endpoint_test.go:20]: Same as above

**Comment by Mehrbod2002** [comment]
@TwiN what is that failure for ?

**Comment by TwiN** [comment]
I think you may have forgotten to update the alerting providers themselves - you only updated their tests.

**Review by TwiN — CHANGES_REQUESTED** [review]
You haven't addressed my last comment - you've mostly made changes in the test, but you haven't modified how the client configuration is used outside of the tests.

For instance, the discord provider is passing a nil client configuration in https://github.com/Mehrbod2002/gatus/blob/7d3b7711c838aebb8ddb226aed0f7c6f0c4bc592/alerting/provider/discord/discord.go#L54

**Comment by Mehrbod2002** [comment]
@TwiN Sorry I got confused .
Did you mean this ? 
Please guide me if it's incorrect as I am a new developer :) Thanks

**Pull request closed without merging**